TB Analysis pipeline
===

* **Quality inspection** [fastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
* **Variant calling** [Pipeline](https://github.com/xiaeryu/TB_pipeline/blob/master/VariantCalling.sh)
* **Consensus building** [Pipeline](https://github.com/xiaeryu/TB_pipeline/blob/master/VariantCalling.sh)
* **Strain typing** [62 SNP markers](https://github.com/xiaeryu/TB_pipeline/blob/master/Scheme_62) & [Script](https://github.com/xiaeryu/TB_pipeline/blob/master/Barcoding.py)
* **Spoligotyping** [SpoTyping](https://github.com/xiaeryu/SpoTyping/releases/) or [SpolPred](http://pathogenseq.lshtm.ac.uk/spolpred)
* **RD analysis** [RD-Analyzer](https://github.com/xiaeryu/RD-Analyzer)
